VideoApp/VideoApp/Stores/Auth/InternalAuthStore.swift

```diff
@@ -58,17 +60,25 @@ class FirebaseAuthStore: NSObject, FirebaseAuthStoreWriting {
     }
 
     func openURL(_ url: URL) -> Bool {
-        return googleSignIn.handle(url)
+        googleSignIn.handle(url)
     }
-    
-    func fetchAccessToken(completion: @escaping (String?, Error?) -> Void) {
-        guard let user = firebaseAuth.currentUser else { completion(nil, nil); return }
+
+    func refreshAccessToken(completion: @escaping () -> Void) {
```

Comment:
I think you understand everything correctly. 

So I called this `accessToken` because I was thinking conventional OAuth and wanted to use same terminology that people expect. Normally a successful auth request results in a response that contains `access_token` and then that is what is used inside the `authorization` header field. But looking at Firebase more I realize that we are using the `id_token` from the auth response. Here is a good example of an auth response that contains both:

https://www.oauth.com/oauth2-servers/signing-in-with-google/getting-an-id-token/

`id_token` has some user info like email encoded in addition to the actual auth secret.

So I think I'm good with renaming `accessToken` to `idToken` because it is a standard term, is not specific to Firebase auth implementation, and that really is how we are using it. At the moment it seems best to stick with the established `idToken` term instead of invent something new. 

So I think this all just works out now. If we were actually using an access token I would probably prefer sticking with that name. Would be afraid it would cause confusion to not call things what they are. But we aren't! 😄 

Glad we looked into this.

Reply:
✅